File: socialmediaagent-api/schemas/company.py
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class CompanyCreate(BaseModel):
    name: str
    website: str | None = None
    description: str | None = None
    industry: str | None = None
    target_audience: str | None = None
    value_proposition: str | None = None
    differentiators: str | None = None
    key_messages: list[str] | None = None


class CompanyUpdate(BaseModel):
    name: str | None = None
    website: str | None = None
    description: str | None = None
    industry: str | None = None
    target_audience: str | None = None
    value_proposition: str | None = None
    differentiators: str | None = None
    key_messages: list[str] | None = None


class CompanyResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    user_id: UUID
    name: str
    website: str | None = None
    description: str | None = None
    industry: str | None = None
    target_audience: str | None = None
    value_proposition: str | None = None
    differentiators: str | None = None
    key_messages: list[str] | None = None

File: socialmediaagent-ui/lib/validations/company.ts
import { z } from "zod";

export const companyCreateSchema = z.object({
  name: z.string().min(2),
  website: z.string().url().optional().or(z.literal("")),
  description: z.string().optional(),
  industry: z.string().optional(),
  target_audience: z.string().optional(),
  value_proposition: z.string().optional(),
  differentiators: z.string().optional(),
  key_messages: z.array(z.string()).optional(),
});

export const companyUpdateSchema = companyCreateSchema.partial();
